File: src/ScribeNest.Domain/Entities/Post.cs
namespace ScribeNest.Domain.Entities;

public class Post
{
    public int Id { get; set; }
    public string Title { get; set; } = "";
    public string Slug { get; set; } = "";
    public string Content { get; set; } = "";
    public DateTime PublishedAt { get; set; } = DateTime.UtcNow;

    public int CategoryId { get; set; }
    public Category? Category { get; set; }

    public ICollection<Comment> Comments { get; set; } = new List<Comment>();
}

File: src/ScribeNest.Front/src/app/core/models/post.ts
export interface PostListItem {
  id: number;
  title: string;
  slug: string;
  excerpt: string;
  categoryId: number;
  category: string;
  publishedAt: string;
  tags: string[];
}

export interface PostDetail extends PostListItem {
  content: string;
}

export interface PostUpsert {
  title: string;
  slug: string;
  content: string;
  tags: string;
  categoryId: number | null;
}

export interface AiSuggestionRequest {
  title: string;
  content: string;
  category?: string;
  tags?: string;
}

export interface AiSuggestionResponse {
  suggestedTitle: string;
  summary: string;
  excerpt: string;
  suggestedTags: string[];
  explainForJuniors: string;
  warnings: string[];
  editorialConsistency: string;
  editorialNotes: string[];
}
